selfdrive/ui/qt/widgets/cameraview.cc

```diff
@@ -292,8 +295,19 @@ void CameraViewWidget::vipcThread() {
       emit vipcThreadConnected(vipc_client.get());
     }
 
-    if (VisionBuf *buf = vipc_client->recv(nullptr, 1000)) {
+    // Wait until camera frame id is behind modelV2, then update once
+    while (meta_main.frame_id >= (*s->sm)["modelV2"].getModelV2().getFrameId()) {
+      qDebug() << "Sleeping";
+      QThread::msleep(5);
```

Comment:
There has to be a better way than polling.

Reply:
yeah not happy with this just yet, will try to use the signals available to the onroad widget. But we still probably want to show the latest camera frames if the model timed out